draft-leon-distributed-multi-signer-00.md

```diff
@@ -444,45 +484,155 @@ has been done, the local MSA is able to initiate communication with
+An MSA signals its synchronization capabilities by setting the corresponding
+bits to 1.
 
+0: Leader/Follower synchronization supported (baseline, MUST be supported by all MSAs)
+
+1: Peer-to-Peer synchronization supported
+
+2: <unused>
+
+3: <unused>
+
+4: <unused>
+
+5: <unused>
+
+6: <unused>
+
+7: <unused>
+
+# Sequence Diagram Example of Establishing Secure Comms - "The Hello Phase"
+The procedure of locating another MSA and establishing a secure communication,
+referred to as "The Hello Phase" is examplified in the sequence diagram below.
+
+The procedure is as follows:
+
+1. The multisigner agents retrieve a zone that is to be signed, the become
+   aware of this from observing the MSIGNER RRset.
+
+2. They start querying any unrecognized MSIGNER identities. Here we only
+   illustrate the baseline case where DNS-based communications is to be used
+   in the following phase.
+
+3. Once an MSA has received the desired responses (SVCB and KEY records in the
+   baseline case) it send a NOTIFY message with a dedicated Multi-Signer OPT
+   code with a "hello" signal. The sender uses this OPT field to signal its
+   transport and synchronization capabilities. Similarly, the responder signals
+   its capabilities using the same field.
+
+4. When an MSA either gets a NOERROR response to its NOTIFY OPT(hello) message
+   or responds with a NOERROR, it transitions out of "The Hello Phase" with
+   the exchanging party and they transition to the next phaste where they start
+   sending NOTIFY OPT(heartbeat) signals instead.
+
+In case one MSA is too fast for the other, perhaps the zone transfer was
+delayed for one of them, the slower one can simply respond in the negative to
+any NOTIFY OPT(hello) it receives (perhaps the other party is polling it). Once
+it is ready, it simply sends a NOTIFY OPT(hello) of its own and will hopefully
+get a NOERROR or, it responds positively to the other party's message.
+
+
+~~~
++----------+                 +----------+                        +----------+
+|  Owner   |                 |  MSA A   |                        |  MSA B   |
++----------+                 +----------+                        +----------+
+     |                            |                                    |
+     |      AXFR(sign-me.se)      |                                    |
+     |--------------------------->|                                    |
+     |      AXFR(sign-me.se)      |                                    |
+     |---------------------------------------------------------------->|
+     |                            |                                    |
+     |                            |                                    |
+     |                            |       QUERY dns.msa-b.se SVCB?     |
+     |                            |----------------------------------->|
+     |                            |       QUERY dns.msa-b.se KEY?      |
+     |                            |----------------------------------->|
```

Comment:
First is the query for _dns._tcp.msa-b.se. URI. The response provides input to the subsequent qnames of the SVCB and KEY queries.